hyperspy/io.py

```diff
@@ -103,45 +102,52 @@ def load(filenames=None,
         If an axis with this name already
         exists it automatically append '-i', where `i` are integers,
         until it finds a name that is not yet in use.
-    mmap: bool
-        If True and stack is True, then the data is stored
-        in a memory-mapped temporary file.The memory-mapped data is
-        stored on disk, and not directly loaded into memory.
-        Memory mapping is especially useful for accessing small
-        fragments of large files without reading the entire file into
-        memory.
-    mmap_dir : string
-        If mmap_dir is not None, and stack and mmap are True, the memory
-        mapped file will be created in the given directory,
-        otherwise the default directory is used.
-    load_to_memory: bool
-        for HDF5 files, blockfiles and EMD files, if True (default) loads all
-        data to memory. If False, enables only loading the data upon request
-    mmap_mode: {'r', 'r+', 'c'}
-        Used when loading blockfiles to determine which mode to use for when
-        loading as memmap (i.e. when load_to_memory=False)
+    lazy : {None, bool}
+        Open the data lazily - i.e. without actually reading the data from the
+        disk until required. Allows opening arbitrary-sized datasets.
+        If None, default from preferences is used.
 
     print_info: bool
         For SEMPER unf- and EMD (Berkley)-files, if True (default is False)
-        additional information read during loading is printed for a quick overview.
-
+        additional information read during loading is printed for a quick
+        overview.
 
     Returns
     -------
     Signal instance or list of signal instances
+
     Examples
     --------
     Loading a single file providing the signal type:
     >>> d = hs.load('file.dm3', signal_type="EDS_TEM")
 
     Loading multiple files:
+
     >>> d = hs.load('file1.dm3','file2.dm3')
+
     Loading multiple files matching the pattern:
+
     >>> d = hs.load('file*.dm3')
+
+    Loading (potentially larger than the available memory) files lazily and
+    stacking:
+
+    >>> s = hs.load('file*.blo', lazy=True, stack=True)
+
     """
+    deprecated = ['load_to_memory', 'mmap', 'mmap_dir', 'mmap_mode']
+    warn_str = "'{}' argument is deprecated, please use 'lazy' instead"
+    for k in deprecated:
+        if k in kwds:
+            lazy=True
+            warnings.warn(warn_str.format(k), VisibleDeprecationWarning)
+            del kwds[k]
```

Comment:
`dask` itself happily writes into anything that supports numpy-style slicing, however since all dask arrays (and hence lazy signals) are immutable, the writing has to be called explicitly.

Currently there is no in-built way to do this, but one could do `s.data.store(mmap)`, which would compute the data and write to `mmap`.

I'll make the changes to allow these modes.

Another cat in the bag is the hdf5 `load_to_memory=True` if there are arrays in the metadata (e.g. a model is stored). Then `deepcopy()` fails, because an open `h5py` handle cannot be copied. That may not be a problem if I wrapped those to dask arrays as well (I have not tested this yet), but it touches things outside the main scope of the PR